reproducibility/codalab/util/analysis_utils.py

```diff
@@ -0,0 +1,124 @@
+    log_types = ["eval"]
+
+    if not splits:
+        splits = DEFAULT_SPLITS
+
+    if include_algorithm_logs:
+        log_types.append("algo")
+
+    for split in splits:
+        for log_type in log_types:
+            log_csv_path = os.path.join(log_url, f"{split}_{log_type}.csv")
+
+            if include_in_distribution and split != "train":
+                # Not all datasets have a ID test and val, so just print a
+                # message if the file does not exist
+                try:
+                    id_log_file = f"id_{split}_{log_type}"
+                    result_dfs[id_log_file] = pd.read_csv(
+                        os.path.join(log_url, f"{id_log_file}.csv")
+                    )
+                except:
+                    print(f"{id_log_file} not available at {log_url}.")
+
+            # Read the csv file and store in the results dataframe
+            result_dfs[f"{split}_{log_type}"] = pd.read_csv(log_csv_path)
+    return result_dfs
+
+
+def compile_results(dataset, results, in_distribution_val=False):
+    metrics = get_metrics(dataset)
+    print(f"Early stopping with {metrics}...")
+
+    compiled_results = dict()
+    for algorithm, result_dfs in results.items():
+        compiled_results[algorithm] = dict()
+        print(f"\nDataset={dataset}, Algorithm={algorithm}")
+
+        for result_type in RESULT_TYPES:
+            if len(result_dfs) > 0 and result_type in result_dfs[0]:
+                compiled_results[algorithm][result_type] = dict()
+                for metric in metrics:
+                    compiled_results[algorithm][result_type][metric] = []
+
+        for result_df in result_dfs:
+            for result_type in RESULT_TYPES:
+                if result_type not in result_df:
+                    continue
+
+                for metric in metrics:
+                    val_split = "id_val_eval" if in_distribution_val else "val_eval"
+                    result = get_early_stopped_row(
+                        result_df[result_type],
+                        result_df[val_split],
+                        sort_metrics=metrics,
+                    )
+                    compiled_results[algorithm][result_type][metric].append(
+                        result[metric]
+                    )
+
+        for result_type in RESULT_TYPES:
+            if result_type in compiled_results[algorithm]:
+                for metric in metrics:
+                    # Take the sample standard deviation (delta degree of freedom = 1)
+                    # and mean across replicates for each metric.
+                    compiled_results[algorithm][result_type][f"{metric}_std"] = np.std(
+                        compiled_results[algorithm][result_type][metric], ddof=1
+                    )
+                    compiled_results[algorithm][result_type][metric] = np.mean(
+                        compiled_results[algorithm][result_type][metric]
+                    )
+    return compiled_results
+
+
```

Comment:
I used this util function to get the early stopped row. Some of the logic in this file is copied from the analysis scripts.